### PR TITLE
add update hook

### PR DESCRIPTION
**Pull request by hellolijj**
### Ⅰ. Describe what this PR does
add update hook 

### Ⅲ. List the added test cases (unit test/integration test) if any, please explain if no tests are needed.
no

### Ⅳ. Describe how to verify it
no

### CODE REVIEW

**Review by PualrDwade — DISMISSED** [review]

File: pkg/controller/cloneset/core/api.go [hunk 1]
```diff
@@ -41,6 +41,7 @@ type Control interface {
 	IsPodUpdateReady(pod *v1.Pod, minReadySeconds int32) bool
 	GetPodsSortFunc(pods []*v1.Pod, waitUpdateIndexes []int) func(i, j int) bool
 	GetUpdateOptions() *inplaceupdate.UpdateOptions
+	ExecUpdateHook(cs *appsv1alpha1.CloneSet, pods []*v1.Pod) bool
```
Comment (FillZpp) [inline pkg/controller/cloneset/core/api.go:44]: Is `ExtraStatusCalculation` better?

File: pkg/controller/cloneset/cloneset_status.go [hunk 2]
```diff
@@ -43,7 +44,8 @@ type realStatusUpdater struct {
 
 func (r *realStatusUpdater) UpdateCloneSetStatus(cs *appsv1alpha1.CloneSet, newStatus *appsv1alpha1.CloneSetStatus, pods []*v1.Pod) error {
 	r.calculateStatus(cs, newStatus, pods)
-	if !r.inconsistentStatus(cs, newStatus) {
+	isChange := clonesetcore.New(cs).ExecUpdateHook(cs, pods)
```
Comment (FillZpp) [inline pkg/controller/cloneset/cloneset_status.go:47]: isChanged

File: pkg/controller/cloneset/core/api.go (hunk 1 above)
Comment (hellolijj) [reply pkg/controller/cloneset/core/api.go]: done!

File: pkg/controller/cloneset/cloneset_status.go (hunk 2 above)
Comment (hellolijj) [reply pkg/controller/cloneset/cloneset_status.go]: done!

**Review by FillZpp — APPROVED** [review]
LGTM

**Pull request merged**